src/custom/pages/Claim/index.tsx

```diff
@@ -174,7 +179,14 @@ export default function Claim() {
 
     // properly reset the user to the claims table and initial investment flow
     resetClaimUi()
-  }, [account, activeClaimAccount, resolvedAddress, isSearchUsed, setActiveClaimAccount, resetClaimUi])
+  }, [account, isSearchUsed, resetClaimUi, setActiveClaimAccount])
+
+  // handle account disconnect or account change after claim is confirmed
+  useEffect(() => {
+    if (!account || (account !== previousAccount && claimStatus === ClaimStatus.CONFIRMED)) {
```

Comment:
shouldn't the useEffect trigger on account change? why is previousAccount needed? I would think
```ts
useEffect(() => {
    if (!account || claimStatus === ClaimStatus.CONFIRMED) {
...
```
should work, it doesn't?

Reply:
Since `claimStatus` is dependency this `(!account || claimStatus === ClaimStatus.CONFIRMED)` would activate instantly when the status becomes confirmed and you would not see claim success view at all. So instead this `account !== previousAccount && claimStatus === ClaimStatus.CONFIRMED` this will only activate when the account changes and the claim is confirmed.